src/MoBi.Presentation/Mappers/AbstractDataTableProviderToQuantityDTOMapper.cs

```diff
@@ -36,16 +36,32 @@ protected AbstractDataTableProviderToQuantityDTOMapper(IMoBiDimensionFactory dim
       ///    Gets a dimension unit string in a data row. Indicate which row contains the unit
       /// </summary>
       /// <param name="rowIndex">The index of the row being converted</param>
-      /// <param name="columnIndex">The index of the column containing the unit</param>
+      /// <param name="unitColumn">The index of the column containing the unit</param>
+      /// <param name="dimensionColumn">The index of the column containing the dimension</param>
       /// <param name="table">The table containing the row</param>
       /// <returns>A dimension corresponding to the unit given</returns>
-      protected IDimension GetDimension(DataTable table, int rowIndex, int columnIndex)
+      protected IDimension GetDimension(DataTable table, int rowIndex, int unitColumn, int dimensionColumn)
       {
          var row = table.Rows[rowIndex];
-         var dimension = _dimensionFactory.DimensionForUnit(row[columnIndex].ToString());
+         var unitName = row[unitColumn].ToString();
+         var dimension = _dimensionFactory.DimensionForUnit(unitName);
          if (dimension == null)
-            throw new ImportQuantityDTOsFromDataTablesMapperException(row, rowIndex, AppConstants.Exceptions.CouldNotFindDimensionFromUnits(row[columnIndex].ToString()));
-         return dimension;
+            throw new ImportQuantityDTOsFromDataTablesMapperException(row, rowIndex,
+               AppConstants.Exceptions.CouldNotFindDimensionFromUnits(unitName));
+
+         if (table.Columns.Count <= dimensionColumn)
+            return dimension;
+
+         //now we may have a dimension specified
+         var dimensionName = row[dimensionColumn].ToString();
+         if (string.IsNullOrEmpty(dimensionName))
+            return dimension;
+
+         if (!_dimensionFactory.Has(dimensionName))
```

Comment:
@Yuri05 Idea is pretty simple.. Same as before. But if a dimension column was specified, we use this instead.
If the dimension does not match the unit, the importer will throw an error as well so all is covered